cmd/kpod/login.go

```diff
@@ -57,6 +63,9 @@ func loginCmd(c *cli.Context) error {
 	}
 
 	sc := common.GetSystemContext("", c.String("authfile"))
+	// set OCICert and DockerCert Path
+	certsdir := filepath.Join(CertsDir, server)
+	sc.OCICertPath, sc.DockerCertPath = certsdir, certsdir
```

Comment:
I think setting the OCICertPath and DockerCertPath to the same location is probably good for now.  However I wonder if the DockerCertPath should be set by Docker somehow and we should use that as a secondary lookup if there are no certs in OCI's location or they're invalid?

@mtrmac thoughts?  I'm on the fence.

Also this change creates the default certificate directory, but does doing this allow 'kpod pull' and 'kpod push' and others to use this directory without specifying the directory at the command line?  I think so via the code in /github.com/containers/image/oci/layout/oci_src.go, but would like to know if you tested that.

Reply:
(Both `OCICertPath` and `DockerCertPath` use the same format, so _that_ is OK.)

> However I wonder if the DockerCertPath should be set by Docker somehow and we should use that as a secondary lookup if there are no certs in OCI's location or they're invalid?

`c/image` automatically uses `DockerPerHostCertDirPath/$server`, i.e. `/etc/docker/certs.d/$server` if `DockerCertPath` is not configured; the above essentially disables `/etc/docker/certs.d` (or should I say “breaks it“?).

OTOH `c/image` uses only the system default CAs for OCI unless `OCICertPath` is explicitly specified.

Overall, the description of this PR is unclear to me, I probably lack the assumed context. “Change the default Certs directory” — the default _for what_? _Why change it at all_?

Reply:
The current default certs directory for kpod is the same as it is for Docker, i.e. /etc/docker/certs.d.  This change is primarily meant to move that default to /etc/containers/cert.d/.  My questioning was is would it be valid to first check /etc/containers/cert.d and if the certificate was not found/invalid, then check /etc/docker/certs.d.  That's why I"m concerned about setting both sc.OCICertPath and sc.DockerCertPath to the same directory.

Reply:
DockerCertPath should stay /etc/docker/certs.d

Reply:
@mtrmac The goal is not to require /etc/docker/certs.d but be independent of it it.  I agree we probably should set a default in containers/image for this with the fallback to the Docker default. Then everyone can use it.

Reply:
@TomSweeneyRedHat  you guessed it right, right now by changing the defaults things seem to break

```console
$ sudo /usr/local/bin/kpod pull docker-registry.engineering.redhat.com/surajd-images/busybox:1.27.2
Trying to pull docker-registry.engineering.redhat.com/surajd-images/busybox:1.27.2...
Failed
error pulling image from "docker-registry.engineering.redhat.com/surajd-images/busybox:1.27.2": Error initializing image from source docker://docker-registry.engineering.redhat.com/surajd-images/busybox:1.27.2: pinging docker registry returned: Get https://docker-registry.engineering.redhat.com/v2/: x509: certificate signed by unknown authority
```

which is failing


to make it work:

```console
$ sudo /usr/local/bin/kpod pull --cert-dir /etc/containers/certs.d/docker-registry.engineering.redhat.com \
 --authfile /run/user/1000/containers/auth.json  \
docker-registry.engineering.redhat.com/surajd-images/busybox:1.27.2                            
Trying to pull docker-registry.engineering.redhat.com/surajd-images/busybox:1.27.2...
Getting image source signatures
Copying blob sha256:0ffadd58f2a61468f527cc4f0fc45272ee4a1a428abe014546c89de2aa6a0eb5
 698.53 KB / 698.53 KB [====================================================] 1s
Copying blob sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4
 32 B / 32 B [==============================================================] 0s
Writing manifest to image destination
Storing signatures
```